### ISSUE TITLE
Port twisted.python.randbytes to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6061]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6061|
|Type|enhancement|
|Created|2012-10-04 19:31:53Z|
|Branch|https://github.com/twisted/twisted/tree/randbytes-py3-6061|

This is needed by `twisted.names.dns`.

<details><summary>Searchable metadata</summary>

```
trac-id__6061 6061
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_randbytes_py3_6061 branches/randbytes-py3-6061
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349379113000000 1349379113000000
changetime__1349452595000000 1349452595000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [35956]) Branching to 'randbytes-py3-6061'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @itamarst|
|-|-|

Okay, worse than I thought, and the module is somewhat useless now that its alternative-selection features have been deleted.  Oh well, I did it anyway, and I wish it had been easier and more useful.  Build started: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/randbytes-py3-6061

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst commented|
|-|-|

I was actually going to suggest (in a new ticket) the use of randrange elsewhere in `twisted.names.dns` be switched to something secure based on inputs from `os.urandom`. Probably `randbytes` would be a reasonable place to add that sort of thing, so might be worth keeping around.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

What do you mean "switched to something secure"?  That's the point of `twisted.python.randbytes.secureRandom`, which is what `twisted.names.dns` currently uses.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst commented|
|-|-|

`DNSMixin.__init__` uses randrange; not sure if that id has any meaning or use though.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Ah.  Good catch!  That `id` looks unused to me.  Perhaps I'll delete it.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_7"></a>|@itamarst set owner to @exarkun|
|-|-|

Looks good, please merge. 

(We should keep our eye out and see if we want to make a `twisted.python.compat.bytechr` if this comes up again.)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set status to `closed`|
|-|-|

(In [35963]) Merge randbytes-py3-6061

Author: exarkun
Reviewer: itamarst
Fixes: [#6061](https://github.com/twisted/twisted/issues/6061)

Port `twisted.python.randbytes` to Python 3.
